zap/src/main/java/org/parosproxy/paros/extension/encoder/Encoder.java

```diff
@@ -123,6 +124,18 @@ public String getBase64Decode(String msg) throws IllegalArgumentException, IOExc
         return new String(Base64.decode(msg, Base64.NO_OPTIONS), base64Charset);
     }
 
+    public String getBase64urlEncode(String msg) throws NullPointerException, IOException {
```

Comment:
I think better if we can check null in the start of the method by using  ``` Objects.requiresNonNull or null comparison ```  also I think there is no need to explicitly stating Null Pointer Exception in throws.

Reply:
I was following the style of the existing code. I don't mind changing it, but I think it would be better to change all these instances instead of just this new one.

Reply:
IMHO, I think we should not introduce something if we are planning to refactor that. We will refactor existing later. @kingthorin and @thc202 please suggest.

Reply:
I don't think worth refactor/tweak these classes, the plan is to drop them at some point, superseded by the new encoder/decoder add-on (yes, the current implementation is still using the `Encoder` class but hopefully that will be changed...).

Reply:
Resolving per above comment.